test/unit-test/rtp_api/rtp_api_utest.c

```diff
@@ -0,0 +1,892 @@
+/* Unity includes. */
+#include "unity.h"
+#include "catch_assert.h"
+
+/* Standard includes. */
+#include <string.h>
+#include <stdint.h>
+
+/* API includes. */
+#include "rtp_api.h"
+#include "rtp_data_types.h"
+#include "rtp_endianness.h"
+
+
+/* ===========================  EXTERN VARIABLES  =========================== */
+
+#define MAX_FRAME_LENGTH        10 * 1024
+#define RTP_HEADER_MIN_LENGTH   12 /* No CSRC and no extension. */
+
+uint8_t buffer[ MAX_FRAME_LENGTH ];
+
+void setUp( void )
+{
+    memset( &( buffer[ 0 ] ),
+            0,
+            sizeof( buffer ) );
+}
+
+void tearDown( void )
+{
+}
+
+/* ==============================  Test Cases  ============================== */
+
+/**
+ * @brief Validate Rtp_Init in case of valid inputs.
+ */
+void test_Rtp_Init_Pass( void )
+{
+    RtpResult_t result;
+    RtpContext_t ctx = { 0 };
+
+    result = Rtp_Init( &ctx );
+
+    TEST_ASSERT_EQUAL( RTP_RESULT_OK,
+                       result );
+
+    TEST_ASSERT_NOT_NULL( ctx.readWriteFunctions.readUint32Fn );
+
+    TEST_ASSERT_NOT_NULL( ctx.readWriteFunctions.writeUint32Fn );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate RTP_Serialize in case of valid inputs.
+ */
+void test_Rtp_Serialize_Pass( void )
+{
+    RtpResult_t result;
+    RtpContext_t ctx = { 0 };
+    RtpPacket_t packet = { 0 };
+    uint8_t buffer[ MAX_FRAME_LENGTH ] = { 0 };
```

Comment:
buffer is declared both globally and locally. Could you name it based on its usage?

Reply:
I have used the local buffer only in my test cases, so removing the global buffer's declaration.